controllers/bpfman-agent/common.go

```diff
@@ -633,6 +633,31 @@ func (r *ReconcilerCommon) handleProgDelete(
 	return internal.Unchanged, nil
 }
 
+func (r *ReconcilerCommon) unloadAndDeleteApplicationProgramsList(ctx context.Context, bpfDeletedPrograms *bpfmaniov1alpha1.BpfProgramList) error {
+	for _, bpfProgram := range bpfDeletedPrograms.Items {
+		r.Logger.V(1).Info("Deleting bpfProgram", "Name", bpfProgram.Name)
+		id, err := bpfmanagentinternal.GetID(&bpfProgram)
+		if err != nil {
+			r.Logger.Error(err, "Failed to get bpf program ID")
+			return err
+		}
+		r.Logger.Info("Calling bpfman to unload program on node", "bpfProgram Name", bpfProgram.Name, "Program ID", id)
+		if err := bpfmanagentinternal.UnloadBpfmanProgram(ctx, r.BpfmanClient, *id); err != nil {
+			r.Logger.Error(err, "Failed to unload Program")
+			return err
+		}
+
+		r.removeFinalizer(ctx, &bpfProgram, internal.BpfApplicationControllerFinalizer)
+
+		opts := client.DeleteOptions{}
+		r.Logger.Info("Deleting bpfProgram", "Name", bpfProgram.Name, "Owner", bpfProgram.GetName())
+		if err := r.Delete(ctx, &bpfProgram, &opts); err != nil {
+			return fmt.Errorf("failed to delete bpfProgram object: %v", err)
+		}
```

Comment:
We would return here too on success.

Reply:
I have some ideas about changing that behavior, but I think we should keep it as is.  The reason is that as soon as you make a change, the manager calls Reconcile() again, and if we don't stop, we end up with a Reconcile storm.  That's the theory anyway.

Reply:
so u wanted to delete one program at a time that is why the return ?

Reply:
That's what ends up happening, but I'd like to look at changing that behavior.

Reply:
yeah this behavior isn't great I watched app sample configs with 6 hooks takes ~30s to complete